src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -288,6 +298,22 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
                     (repository.Endpoint.Id == Guid.Empty && string.Equals(x.Name, repository.Endpoint.Name.ToString(), StringComparison.OrdinalIgnoreCase)));
             }
 
+            bool isForkedBuild = StringUtil.ConvertToBoolean(executionContext.Variables.GetValueOrDefault("System.PullRequest.IsFork")?.Value);
+            executionContext.Debug($"isForkedBuild: {isForkedBuild}");
+            // use the SystemVssConnection for Azure Repos forked builds
+            if(endpoint == null && isForkedBuild && UseSystemVssConnectionForForkedBuilds())
```

Comment:
Why doesn't the server simply provide the correct endpoint for this repository? I don't think the agent should be doing special work here.